File: 프로그래머스/1/12933. 정수 내림차순으로 배치하기/정수 내림차순으로 배치하기.py
def solution(n):
    strlist = list(str(n))
    strlist = sorted(strlist, key = lambda x : -int(x))
    size = len(strlist) - 1
    answer = 0
    for s in strlist:
        if size == 0:
            answer += int(s)
        else:
            answer += int(s) * (10 ** size)
        size -= 1
    return answer

File: 프로그래머스/1/12933. 정수 내림차순으로 배치하기/정수 내림차순으로 배치하기.java
import java.lang.StringBuilder;
import java.util.Arrays;


class Solution {
    public long solution(long n) {
        long answer = 0;
        String[] str = String.valueOf(n).split("");
        Arrays.sort(str);
        
        StringBuilder sb = new StringBuilder();
        for (String s : str) {
            sb.append(s);
        }
        
        answer = Long.parseLong(sb.reverse().toString());
        return answer;
    }
}